spi/data-plane/data-plane-spi/src/test/java/org/eclipse/edc/connector/dataplane/spi/pipeline/StreamFailureTest.java

```diff
@@ -0,0 +1,69 @@
+/*
+ *  Copyright (c) 2024 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG) - Initial implementation
+ *
+ */
+
+package org.eclipse.edc.connector.dataplane.spi.pipeline;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.eclipse.edc.connector.dataplane.spi.pipeline.StreamFailure.Reason.GENERAL_ERROR;
+import static org.eclipse.edc.connector.dataplane.spi.pipeline.StreamFailure.Reason.NOT_AUTHORIZED;
+import static org.eclipse.edc.connector.dataplane.spi.pipeline.StreamFailure.Reason.NOT_FOUND;
+
+public class StreamFailureTest {
+    private StreamFailure streamFailure;
+    private String failureMessage;
+
+    @BeforeEach
+    void setup() {
+        failureMessage = "FAILURE MESSAGE";
+    }
+
+    @Test
+    void verify_with_failureDetail_message_startWith_notFound() {
+
+        streamFailure = new StreamFailure(List.of(failureMessage), NOT_FOUND);
+        assertThat(streamFailure.getFailureDetail()).startsWith(NOT_FOUND.toString());
+    }
+
+    @Test
+    void verify_with_failureDetail_message_startWith_notAuthorized() {
+
+        streamFailure = new StreamFailure(List.of(failureMessage), NOT_AUTHORIZED);
+        assertThat(streamFailure.getFailureDetail()).startsWith(NOT_AUTHORIZED.toString());
+    }
+
+    @Test
+    void verify_with_failureDetail_message_startWith_generalError() {
+
+        streamFailure = new StreamFailure(List.of(failureMessage), GENERAL_ERROR);
+        assertThat(streamFailure.getFailureDetail()).startsWith(GENERAL_ERROR.toString());
+    }
```

Comment:
given that the behavior is the same for all the reasons, please either leave just one test with a reason or use a `ParameterizedTest` with an `EnumSource` that will be easier to read than having a test for every reason.

I would prefer the first, because the behavior is actually really straightforward.